File: 2245-destroying-asteroids/destroying-asteroids.py
class Solution:
    def asteroidsDestroyed(self, mass: int, asteroids: List[int]) -> bool:
        # one line solution! (amazing i know) Somehow beats 85% mem :P
        return (cm:=mass) and (min([(cm := cm + (asteroids[i] if cm >= asteroids[i] else float("-inf"))) for i in range(max(bool(asteroids.sort()),len(asteroids)))]) > 0)

File: 2245-destroying-asteroids/destroying-asteroids.cpp
class Solution {
public:
    bool asteroidsDestroyed(int mass, vector<int>& asteroids) {
        // greedy rahhhhh
        // overflow :(
        long long int m = mass;
        sort(asteroids.begin(), asteroids.end());
        for (int i = 0; i < asteroids.size(); i++) {
            if (m >= asteroids[i]) {
                m += asteroids[i];
            } else return false;
        }
        return true;
    }
};